tests/components/RangeInput.test.tsx

```diff
@@ -0,0 +1,257 @@
+    });
+  });
+
+  it('renders correctly when clearing input and no state is set', async () => {
+    renderRangeInput(filterContextValue);
+    const actions = spyOnActions();
+    const maxTextbox = screen.getAllByRole('textbox')[1];
+    userEvent.type(maxTextbox, '20');
+    await waitFor(() => {
+      expect(maxTextbox).toHaveValue('20');
+    });
+    userEvent.click(screen.getByText('Clear min and max'));
+    expect(maxTextbox).toHaveValue('');
+    expect(actions.setFilterOption).toHaveBeenCalledTimes(0);
+  });
+});
+
+describe('Renders correctly for min and max inputs', () => {
+  beforeEach(() => {
+    mockAnswersHooks({ mockedState, mockedActions });
+  });
+
+  it('renders correctly when inputing range and applies proper values in state', async () => {
+    renderRangeInput(filterContextValue);
+    const actions = spyOnActions();
+    const [minTextbox, maxTextbox] = screen.getAllByRole('textbox');
+    userEvent.type(minTextbox, '10');
+    userEvent.type(maxTextbox, '20');
+    await waitFor(() => {
+      expect(minTextbox).toHaveValue('10');
+    });
+    expect(maxTextbox).toHaveValue('20');
+    expect(screen.getByText('Apply')).toBeDefined();
+    expect(screen.getByText('Clear min and max')).toBeDefined();
+    userEvent.click(screen.getByText('Apply'));
+    expect(actions.setFilterOption).toHaveBeenCalledWith({
+      displayName: '10 - 20',
+      fieldId: '123',
+      matcher: '$between',
+      selected: true,
+      value: {
+        start: {
+          matcher: '$ge',
+          value: 10
+        },
+        end: {
+          matcher: '$le',
+          value: 20
+        },
+      }
+    });
+  });
+
+  it('renders correctly when clearing inputs and no state is set', async () => {
+    renderRangeInput(filterContextValue);
+    const actions = spyOnActions();
+    const [minTextbox, maxTextbox] = screen.getAllByRole('textbox');
+    userEvent.type(minTextbox, '10');
+    userEvent.type(maxTextbox, '20');
+    await waitFor(() => {
+      expect(minTextbox).toHaveValue('10');
+    });
+    expect(maxTextbox).toHaveValue('20');
+    userEvent.click(screen.getByText('Clear min and max'));
+    expect(minTextbox).toHaveValue('');
+    expect(maxTextbox).toHaveValue('');
+    expect(actions.setFilterOption).toHaveBeenCalledTimes(0);
+  });
+
+  it('renders correctly when input range is invalid and no filter is set in state', async () => {
+    renderRangeInput(filterContextValue);
+    const [minTextbox, maxTextbox] = screen.getAllByRole('textbox');
+    userEvent.type(minTextbox, '20');
+    userEvent.type(maxTextbox, '10');
+    const actions = spyOnActions();
+    await waitFor(() => {
+      expect(minTextbox).toHaveValue('20');
+    });
+    expect(maxTextbox).toHaveValue('10');
+    expect(screen.getByText('Invalid range')).toBeDefined();
+    expect(actions.setFilterOption).toHaveBeenCalledTimes(0);
+  });
+});
+
+it('renders correctly when disabled', () => {
+  renderRangeInput(filterContextValueDisabled);
```

Comment:
What do you think about defining `filterContextValueDisabled` within this test since it's only used here? Having a huge block of declared fixtures at the top of the file on one hand could be viewed as organized, on the other hand it's a little overwhelming when you first open things up

Reply:
This is more a matter a taste I think so feel free to ignore?